src/manifest/dbtProject.ts

```diff
@@ -99,7 +96,7 @@ export class Manifest {
         new Map(),
         this.projectRoot
       );
-      this.onManifestCacheChangedHandlers.forEach((handler) => handler(event));
+      dbtProjectContainer.passEventToProviders(event);

```

Comment:
dbtProjectContainer should have static methods like raiseSourceFileChangedEvent and raiseManifestChangedEvent.

Within the dbtProject we don't care to whom the event is raised, so we don't need to know that only the providers are interested in this event.